Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +659,157 @@ class InAppMessageViewTest: IntegrationTest {
+
+    @MainActor
+    func test_onInlineMessageShown_givenDelegateNotSet_onSingleButtonTap_expectTrackClickedMetric_expectGlobalMessageActionTakenCallback() async {
+        DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
+        let inlineView = await showInlineMessageForMetrics()
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Tap button once
+        onCustomActionButtonPressed(onInlineView: inlineView)
+
+        // Check if messageActionTaken is called
+        XCTAssertTrue(eventListenerMock.messageActionTakenCalled)
+        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 1)
+
+        // Check other handlers are not called
+        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
+        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        // The first post call occurs when the message is shown,
+        // and the second post call is triggered by the button click action.
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 2)
+    }
+
+    @MainActor
+    func test_onInlineMessageShown_givenDelegateNotSet_onMultipleButtonTap_expectGlobalMessageActionTakenCallback() async {
+        DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
+        let inlineView = await showInlineMessageForMetrics()
+        DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+        onCustomActionButtonPressed(onInlineView: inlineView)
+
+        // Check if messageActionTaken is called
+        XCTAssertTrue(eventListenerMock.messageActionTakenCalled)
+        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 1)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        // The first post call occurs when the message is shown,
+        // and the second post call is triggered by the button click action.
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 2)
+
+        // Check other listeners are not called except shown
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
+        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
+
+        // Tap button again
+        onCustomActionButtonPressed(onInlineView: inlineView)
+        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 2)
+        // Tap the button again and verify that
+        // the count of post calls to track metrics updates
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 3)
+    }
+
+    @MainActor
+    func test_onInlineMessageShown_givenDelegateSet_onMultipleButtonTap_expectSingleTrackClickedMetric_expectGlobalMessageActionTakenCallback() async {
```

Comment:
I think this test should be broken into multiple test functions. I also think event listener tests out of scope for this ticket.

Reply:
Broken.
`messageShown` and `messageActionTaken` cases are handled in this PR as per description https://github.com/customerio/customerio-ios/pull/775